vcd/resource_vcd_vapp_vm.go

```diff
@@ -53,26 +59,40 @@ func resourceVcdVAppVm() *schema.Resource {
 				Description: "The name of VDC to use, optional if defined at provider level",
 			},
 			"template_name": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the vApp Template to use",
 			},
 			"catalog_name": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The catalog name in which to find the given vApp Template",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The VM description",
+				// Note: description is read only, as we lack the needed fields to set it at creation.
+				// Currently, this field has the description of the OVA used to create the VM
 			},
 			"memory": &schema.Schema{
-				Type:     schema.TypeInt,
-				Optional: true,
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Description:  "The amount of RAM (in MB) to allocate to the VM",
+				ValidateFunc: validateMultipleOf4(),
```

Comment:
Why is this validate function added? Is this what vCD expects and if yes, why is it a multiple of four and not eight?

Reply:
If we don't have this validation, and the memory is incorrect (for example `1` or `103`) we get an error saying "The virtual machine memory size must be a multiple of 4 MB".
However, the error happens **after the virtual machine has already been created**, leaving the resource in a tainted state.
Using this validation, the failure happens immediately, and users can change the offending value and try again.

Reply:
Which vCD version did you observe this error in?

Reply:
Adding the original comment from community user where the need for this originated for reference:
https://github.com/terraform-providers/terraform-provider-vcd/issues/239#issuecomment-550375527

Reply:
> Which vCD version did you observe this error in?

9.5 and 10.0